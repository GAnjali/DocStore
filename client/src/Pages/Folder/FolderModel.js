import React from "react";

const FolderModel = (props) => {
    let style = {};
    if (props.show === true) {
        style = {display: "block"};
    }
    return (
        <div id="myModal" className="sharingmodal" style={style}>
            <div className="foldermodal-content">
                <div className="modal-header">
                    <span className={"newFolder"}>New Folder</span>
                    <span className="close" onClick={props.handleClose}>&times;</span>
                    <textarea className={"sharingwith-text-area"} name={"newFolderName"}
                              placeholder={"Enter folder name here..."} onChange={props.handleFolderNameChange}/>
                    <div className="modal-">
                        <button onClick={props.handleSaveFolder}>Save</button>
                    </div>
                </div>
            </div>
        </div>
    )
};

export default FolderModel;