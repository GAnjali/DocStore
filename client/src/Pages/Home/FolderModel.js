import React from "react";

const FolderModel = (props) => {
    let style = {}, title = '', content = '';
    if (props.show === true) {
        style = {display: "block"};
    }
    return (
        <div id="myModal" className="sharingmodal" style={style}>
            <div className="sharingmodal-content">
                <div className={"sharingmodel-body"}>
                    <textarea className={"sharewith-textarea"} placeholder={"Enter folder name here..."} onChange={props.handleFolderNameChange}/>
                </div>
                <button className={"share-button"} onClick={props.handleSaveFolder}>Save</button>
            </div>
        </div>
    )
};

export default FolderModel;