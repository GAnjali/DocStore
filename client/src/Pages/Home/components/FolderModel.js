import React from "react";
import {
  FOLDER_NAME_PLACEHOLDER,
  NEW_FOLDER,
  SAVE
} from "../../../AppConstants";

const FolderModel = props => {
  const { show, handleClose, handleSaveFolder, handleFolderNameChange } = props;
  let style = {};
  if (show) {
    style = { display: "block" };
  }
  return (
    <div id="myModal" className="sharingmodal" style={style}>
      <div className="foldermodal-content">
        <div className="modal-header">
          <span className={"newFolder"}>{NEW_FOLDER}</span>
          <span className="close" onClick={handleClose}>
            &times;
          </span>
          <textarea
            className={"sharingwith-text-area"}
            name={"newFolderName"}
            placeholder={FOLDER_NAME_PLACEHOLDER}
            onChange={handleFolderNameChange}
          />
          <div className="modal-save">
            <button onClick={handleSaveFolder}>{SAVE}</button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default FolderModel;