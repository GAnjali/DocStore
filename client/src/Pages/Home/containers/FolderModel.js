import React, { Component } from "react";
import { addFolder } from "../services/FolderService";
import FolderModelComponent from "../components/FolderModel";

class FolderModel extends Component {
  constructor(props) {
    super(props);
    this.state = { newFolderName: "", error: "" };
  }

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSaveFolder = async () => {
    const newFolder = { name: this.state.newFolderName };
    await addFolder(newFolder)
      .then(() => {
        this.props.handleClose();
        this.props.updateComponent(localStorage.getItem("parentfolderid"));
      })
      .catch(err => {
        this.setState({
          error: err.message
        });
      });
  };

  render() {
    const { show, handleClose } = this.props;
    let style = {};
    if (show) {
      style = { display: "block" };
    }
    return (
      <FolderModelComponent
        style={style}
        handleClose={handleClose}
        handleChange={this.handleChange}
        handleSaveFolder={this.handleSaveFolder}
      />
    );
  }
}

export default FolderModel;