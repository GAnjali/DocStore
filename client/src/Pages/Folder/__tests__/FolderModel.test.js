import React from "react";
import {shallow} from "enzyme";
import FolderModel from "../FolderModel";

describe("FolderModel component tests", () => {
    const props = {
        handleFolderNameChange: jest.fn(),
        handleSaveFolder: jest.fn()
    };

    it("Should render without crashing", () => {
        shallow(<FolderModel props/>);
    });

    it('should match the FolderModel snapshot', function () {
        const folderModelComponent = shallow(<FolderModel props/>);
        expect(folderModelComponent).toMatchSnapshot();
    });
});
