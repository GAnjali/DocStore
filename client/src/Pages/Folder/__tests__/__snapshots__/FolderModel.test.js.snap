// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderModel component tests should match the FileModel snapshot 1`] = `ShallowWrapper {}`;
